castai/resource_node_template.go

```diff
@@ -951,10 +951,10 @@ func toTemplateConstraints(obj map[string]any) *sdk.NodetemplatesV1TemplateConst
 	if v, ok := obj[FieldNodeTemplateFallbackRestoreRateSeconds].(int); ok {
 		out.FallbackRestoreRateSeconds = toPtr(int32(v))
 	}
-	if v, ok := obj[FieldNodeTemplateGpu].([]any); ok && len(v) > 0 {
+	if v, ok := obj[FieldNodeTemplateGpu].([]any); ok && len(v) > 0 && v[0] != nil {
```

Comment:
Instead of nil checks I would suggest to use type assertion because it's possible that the value in the slice is still not of type `map[string]any`. Something like this:
```
val, ok := v[0].(map[string]any)
if ok {
    out.Gpu = toTemplateConstraintsGpuConstraints(val)
}
```

Reply:
thx, updated